### DISCUSSION TITLE
[mysql rds] trouble with ssl connect to mysql rds

### DISCUSSION THREAD

**Discussion by gevuong** [prisma/prisma#5038] [Q&A] [answered]
**Environment Setup**
-

```
OS: Catalina 10.15.6
@nestjs/core: 7.5.1
@prisma/client: 2.14.0
@prisma/cli: 2.14.0
```

**Error Description**
-

I'm running `prisma introspect` and am getting the following error:

```
Error: P1000

Authentication failed against database server at `xxxxx.rds.amazonaws.com`, the provided database credentials for `USER` are not valid.

Please make sure to provide valid database credentials for the database server at `xxxxx.rds.amazonaws.com`.
```

I'm using the following database url: `DATABASE_URL="mysql://USER:PASSWORD@xxxxx.rds.amazonaws.com:3306/DATABASE&sslidentity=keyStore.p12&sslpassword=SSLPASS&sslcert=server-ca-cert.pem"`

content in `/prisma/schema.prisma` file:
```
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}

generator client {
  provider = "prisma-client-js"
}
```

**Debugging Steps Taken**
-

* [x] successfully ssl connected using DBeaver (a SQL client tool) with the same credentials, host, port, and CA cert.

* [x] moved CA cert into the `./prisma` directory and tried `prisma introspect` with just the `sslcert` argument in the database url: `DATABASE_URL="mysql://USER:PASSWORD@xxxxx.rds.amazonaws.com:3306/DATABASE&sslcert=server-ca-cert.pem"`

* [x] followed the instructions [here](https://serverfault.com/questions/831394/how-can-i-create-a-pkcs12-file-using-openssl-self-signed-certs) on how to create a `PKCS12` file using openssl:

1. `openssl req -x509 -newkey rsa:4096 -keyout myKey.pem -out cert.pem -days 365` (create client key and client cert)
2. `openssl pkcs12 -export -out keyStore.p12 -inkey myKey.pem -in cert.pem` (create PKCS12 file)

 then moved the `PKCS12` file (`keyStore.p12`) to `./prisma` directory and tried ssl connecting using the following database url: `DATABASE_URL="mysql://USER:PASSWORD@xxxxx.rds.amazonaws.com:3306/DATABASE&sslidentity=keyStore.p12&sslpassword=SSLPASS&sslcert=server-ca-cert.pem"`

They all result in the same `Authentication failed` error. Any advice on debugging this issue?

**Comment by gevuong** [comment, marked as answer]
Update: 

I tried creating the PKCS12 file using the exact steps listed in [Issue #2676](https://github.com/prisma/prisma/issues/2676), then using the following database url: `DATABASE_URL="mysql://USER:PASS@xxx.rds.amazonaws.com:3306/DATABASE&sslidentity=client-identity.p12&sslpassword=SSLPASS&sslcert=server-ca-cert.pem"`, but to no avail..

Hm, I did leave a few fields blank in the Distinguished Name (DN) of the certificate request. But I'm not sure if that would affect the authentication because it probably won't get signed by a CA..(according to [blog](https://msol.io/blog/tech/create-a-self-signed-ssl-certificate-with-openssl/) post.

**Reply by ryands17** [reply]
Hey @gevuong 👋 
This seems like an issue. It would be great if you could create one with the above steps [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=bug_report.md&title=) so that we can look into this 😄

**Reply by gevuong** [reply]
Ok! Created here https://github.com/prisma/prisma/issues/5132.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
